src/javaApi.cpp

```diff
@@ -95,6 +95,36 @@ Java_one_profiler_AsyncProfiler_execute0(JNIEnv* env, jobject unused, jstring co
     return NULL;
 }
 
+extern "C" DLLEXPORT jbyteArray JNICALL
+Java_one_profiler_AsyncProfiler_executeAndGetByteArray0(JNIEnv* env, jobject unused, jstring command) {
+    Arguments args;
+    const char* command_str = env->GetStringUTFChars(command, NULL);
+    Error error = args.parse(command_str);
+    env->ReleaseStringUTFChars(command, command_str);
+
+    if (error) {
+        throwNew(env, "java/lang/IllegalArgumentException", error.message());
+        return NULL;
+    }
+    if (args.hasOutputFile()) {
+        throwNew(env, "java/lang/IllegalArgumentException", "executeAndGetBuffer calls should not specify an output file argument");
```

Comment:
`return` is missing.
Throwing an exception from JNI does not automatically bail out of native method.

Reply:
9866f27bdf4e904aa4581e884d7a051aac48da47